lib/streamlit/elements/map.py

```diff
@@ -157,13 +158,20 @@ def _get_zoom_level(distance: float) -> int:
 
 
 def to_deckgl_json(data: Data, zoom: Optional[int]) -> str:
-    # TODO(harahu): The ignore statement here is because iterables don't have
-    #  the empty attribute. This is either a bug, or the documented data type
-    #  is too broad. One or the other should be addressed, and the ignore
-    #  statement removed.
-    if data is None or data.empty:  # type: ignore[union-attr]
+    if data is None:
         return json.dumps(_DEFAULT_MAP)
 
+    if hasattr(data, "empty"):
+        # TODO(harahu): The ignore statement here is because iterables don't have
+        #  the empty attribute. This is either a bug, or the documented data type
+        #  is too broad. One or the other should be addressed, and the ignore
+        #  statement removed.
+        if data.empty:  # type: ignore[union-attr]
+            return json.dumps(_DEFAULT_MAP)
+
+    if type_util.is_snowpark_dataframe(data):
```

Comment:
I think we actually can just always call `convert_anything_to_df` here without the `is_snowpark_dataframe` check. I don't think there is any reason why `map` should not have the same support for dataframe-like input types as our other elements. In that case, the `data = pd.DataFrame(data)` in line 200 would need to be removed and the docstring would need to be updated to cover the same types as `st.dataframe`.

Reply:
Now we call always `convert_anything_to_df` 👍